### PR TITLE
implicit-casts:false on flutter_tools

### PR DESCRIPTION
**Pull request by a14n**
## Description

Enable `implicit-casts:false` on `flutter_tools`

## Related Issues

None

## Tests

None because it's a refactoring.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart [hunk 1]
```diff
@@ -32,7 +32,7 @@ void main() {
     setUp(() {
       mockProcessManager = mocks.MockProcessManager();
       xcodeProjectInterpreter = XcodeProjectInterpreter();
-      macOS = fakePlatform('macos');
+      macOS = fakePlatform('macos') as FakePlatform;
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart]: Same comment here. Can `fakePlatform` just return a `FakePlatform`?

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart [hunk 2]
```diff
@@ -552,7 +552,7 @@ include ':app'
       fs = MemoryFileSystem();
       mockArtifacts = MockLocalEngineArtifacts();
       mockProcessManager = MockProcessManager();
-      android = fakePlatform('android');
+      android = fakePlatform('android') as FakePlatform;
```
[the remark this answers is not in the document]
Comment (a14n) [reply packages/flutter_tools/test/general.shard/android/gradle_test.dart]: Done

File: packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart (hunk 1 above)
Comment (a14n) [reply packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart]: Done

**Review by zanderso — APPROVED** [review]
lgtm

**Comment by zanderso** [comment]
I'm hearing that this probably needs to be rebased before the presubmit checks will turn green.

**Pull request merged**